plotly_resampler/figure_resampler/figure_resampler_interface.py

```diff
@@ -1283,6 +1290,24 @@ def construct_update_data(
             layout_traces_list.append(trace_reduced)
         return layout_traces_list
 
+    @staticmethod
+    def _parse_dtype_orjson(series: np.ndarray) -> np.ndarray:
+        """Verify the orjson compatibility of the series and convert it if needed."""
+        # NOTE:
+        #    * float16 and float128 aren't supported with latest orjson versions (3.8.1)
+        #    * this method assumes that the it will not get a float128 series
+        if series.dtype in [np.float16]:
+            return series.astype(np.float32)
+
+        # orjson < 3.8.0 encoding cannot encode with int16 & uint16 dtype
+        elif series.dtype in [np.int16, np.uint16]:
+            major_v, minor_v = list(map(int, orjson.__version__.split(".")))[:2]
+            if major_v < 3 or major_v == 3 and minor_v < 8:
+                if series.dtype == np.uint16:
+                    return series.astype("uint32")
+                return series.astype(np.int32)
```

Comment:
Why won't we update the orjson dependency to >= 3.8?
-> This avoids maintaining this code

Reply:
I agree

Reply:
Updated the `poetry.toml`

Reply:
i presume you mean `pyproject.toml` 😉